feature_list/feature_list.cpp

```diff
@@ -0,0 +1,267 @@
+    if (!needsQuotes)
+    {
+      out << value;
+    }
+    else
+    {
+      string quoted(value);
+      size_t pos = 0;
+      while ((pos = quoted.find('"', pos)) != string::npos)
+      {
+        quoted.insert(pos, 1, '"');
+        pos += 2;
+      }
+      out << '"' << quoted << '"';
+    }
+  }
+  out << endl;
+}
+
+class Processor
+{
+  search::ReverseGeocoder m_geocoder;
+  my::Cancellable m_cancellable;
+  search::VillagesCache m_villagesCache;
+  search::LocalityFinder m_finder;
+
+public:
+  Processor(Index const & index)
+    : m_geocoder(index), m_villagesCache(m_cancellable), m_finder(index, m_villagesCache)
+  {
+  }
+  void operator()(FeatureType const & f, uint32_t const & id) { process(f); }
+  void process(FeatureType const & f)
+  {
+    f.ParseBeforeStatistic();
+    string category = GetReadableType(f);
+    if (!f.HasName() || f.GetFeatureType() == feature::GEOM_LINE || category.empty())
+      return;
+    m2::PointD center = FindCenter(f);
+    ms::LatLon ll = MercatorBounds::ToLatLon(center);
+    osm::MapObject obj;
+    obj.SetFromFeatureType(f);
+
+    string city;
+    m_finder.GetLocality(center, city);
+
+    string mwmName = f.GetID().GetMwmName();
+    string name, secondary;
+    f.GetPreferredNames(name, secondary);
+    string uid = BuildUniqueId(ll, name);
+    string lat = strings::to_string_dac(ll.lat, 6);
+    string lon = strings::to_string_dac(ll.lon, 6);
+    search::ReverseGeocoder::Address addr;
+    string address = m_geocoder.GetExactAddress(f, addr)
+                         ? addr.GetStreetName() + ", " + addr.GetHouseNumber()
+                         : "";
+    string phone = f.GetMetadata().Get(feature::Metadata::FMD_PHONE_NUMBER);
+    string cuisine = strings::JoinStrings(obj.GetLocalizedCuisines(), ", ");
+    string opening_hours = f.GetMetadata().Get(feature::Metadata::FMD_OPEN_HOURS);
+    string website = "https://itunes.apple.com/app/id510623322?mt=8";
+
+    vector<string> columns = {uid,  lat,     lon,   mwmName, category,      name,
+                              city, address, phone, cuisine, opening_hours, website};
+    AppendNames(f, columns);
+    PrintAsCSV(columns, cout, ';');
+  }
+};
+
+void PrintHeader()
+{
+  vector<string> columns = {"id",   "lat",     "lon",   "mwm",     "category",      "name",
+                            "city", "address", "phone", "cuisine", "opening_hours", "url"};
+  PrintAsCSV(columns, cout, ';');
+}
+
+void DidDownload(storage::TCountryId const & /* countryId */,
+                 shared_ptr<platform::LocalCountryFile> const & /* localFile */)
+{
+}
+
+bool WillDelete(storage::TCountryId const & /* countryId */,
+                shared_ptr<platform::LocalCountryFile> const & /* localFile */)
+{
+  return false;
+}
+
+int main(int argc, char ** argv)
+{
+  search::ChangeMaxNumberOfOpenFiles(search::kMaxOpenFiles);
+  if (argc <= 1)
+  {
+    LOG(LERROR, ("Usage:", argc == 1 ? argv[0] : "feature_list",
```

Comment:
1) Gflags doesn't provide positional argument handling, so this would complicate execution.
2) Googling for it revealed that some projects were moving away from this library, e.g. chromium.

So no.